app/routes/facebook.js

```diff
@@ -0,0 +1,48 @@
+import Route from '@ember/routing/route';
+
+export default Route.extend({
+  queryParams: {
+    code     : '',
+    provider : ''
+  },
+  model(queryParams) {
+    this.get('loader').post(`/auth/oauth/login/${ queryParams.provider }?code=${ queryParams.code }`)
```

Comment:
All this controlling logic should be inside the controller of the route. Not the model hook.

Reply:
@CosmicCoder96 @niranjan94 Need some help here. I tried making a controller for the route 'oauth/callback' and a js file in controller folder with the same path, and noticed it doesn't get called when the route is accessed.

Reply:
![Screenshot from 2019-03-23 00-00-02](https://user-images.githubusercontent.com/31065687/54845144-aaa73600-4cfe-11e9-9e83-5467aec49e33.png)
Here is the file path.

Reply:
if your route name is facebook then the controller should also be at routes/facebook.

Reply:
@CosmicCoder96 In routes/facebook or controller/facebook?

Reply:
`controller/facebook`